src/rgw/rgw_admin.cc

```diff
@@ -2930,7 +2930,11 @@ int main(int argc, const char **argv)
   if (raw_storage_op) {
     store = RGWStoreManager::get_raw_storage(g_ceph_context);
   } else {
-    store = RGWStoreManager::get_storage(g_ceph_context, false, false, false, false, false);
+    bool run_sync_thread = false;
+    if (opt_cmd == OPT_DATA_SYNC_RUN) {
+      run_sync_thread = true;
```

Comment:
I don't like this as running sync thread has a lot of side effects.
I prefer another fix